app/models/fact_value.rb

```diff
@@ -45,6 +45,15 @@ def self.search_by_host(key, operator, value)
     { :conditions => "fact_values.id IN(#{search.join(',')})" }
   end
 
+  def self.search_by_hostgroup(key, operator, value)
+    search_term = value =~ /\A\d+\Z/ ? 'id' : 'name'
+    conditions = sanitize_sql_for_conditions(["hostgroups.#{search_term} #{operator} ?", value_to_sql(operator, value)])
+    search     = FactValue.joins(:hostgroup).where(conditions).uniq.pluck('fact_values.id')
+
+    return { :conditions => "1=0" } if search.empty?
+    { :conditions => "fact_values.id IN(#{search.join(',')})" }
```

Comment:
Even better, I think this should work just as well and get rid of the extra query completely:
```ruby
search_term = value =~ /\A\d+\Z/ ? 'id' : 'name'
conditions = sanitize_sql_for_conditions(["hostgroups.#{search_term} #{operator} ?", value_to_sql(operator, value)])
{ :joins => :hostgroup, :conditions => conditions }
```